src/DotVVM.Framework/Resources/Scripts/collections/arrayHelper.ts

```diff
@@ -0,0 +1,122 @@
+﻿import { orderBy, orderByDesc } from './sortingHelper'
+
+export {
+    all,
+    any,
+    clear,
+    distinct,
+    firstOrDefault,
+    forEach,
+    lastOrDefault,
+    max,
+    min,
+    orderBy,
+    orderByDesc,
+    remove,
+    removeFirst
+}
+
+function any<T>(source: T[], predicate: (s: T) => boolean): boolean {
+    return firstOrDefault(source, predicate) != null;
+}
+
+function all<T>(source: T[], predicate: (s: T) => boolean) {
+    for (let i = 0; i < source.length; i++) {
+        if (!predicate(source[i])) {
+            return false;
+        }
+    }
+    return true;
+}
+
+function clear<T>(source: T[]): void {
+    source.splice(0, source.length);
+}
+
+function distinct<T>(source: T[]): T[] {
+    let r = [];
+    for (let i = 0; i < source.length; i++) {
+        let found = false;
+        for (var j = 0; j < r.length; j++) {
+            if (r[j] == source[i]) {
```

Comment:
I am not sure about this comparison. I would rather pass a comparer to this function as an optional parameter. In case the comparer is not specified, default compared would be used.

Reply:
Yes, there is no overload with lambda for `Distinct`. Moreover, I added a compilation check that ensures we pass only collections of primitive types. Hopefully, the default comparer should work on primitive types without any problems.

Reply:
I know that there is not an overload that could provide the comparer. In fact, I have meant that when this is a public JS API, it would be nice to have the option to use also comparer. I think we can keep the `==` comparison for now. If there is a use-case for a more sophisticated comparison then I would add it.